sound/soc/intel/boards/Kconfig

```diff
@@ -500,6 +500,21 @@ config SND_SOC_INTEL_SOF_CS42L42_MACH
 	   Say Y if you have such a device.
 	   If unsure select "N".
 
+config SND_SOC_INTEL_SOF_AMP_ONLY_MACH
+	tristate "SOF with speaker amplifier only"
+	depends on I2C && ACPI
+	depends on ((SND_HDA_CODEC_HDMI && SND_SOC_SOF_HDA_AUDIO_CODEC) &&\
+		    (MFD_INTEL_LPSS || COMPILE_TEST))
+	select SND_SOC_CS35L41_I2C
+	select SND_SOC_DMIC
+	select SND_SOC_HDAC_HDMI
+	select SND_SOC_INTEL_HDA_DSP_COMMON
+	help
+	   This adds support for ASoC machine driver for SOF platforms
+	   with speaker amplifier only.
+	   Say Y if you have such a device.
+	   If unsure select "N".
+
```

Comment:
I wonder if we could reuse the work already done for sof_rt1308.c, where there's also no headset codec.

It's the same case of an amplifier, the differences are that you have BT offload and they have HDMI inputs.

Still very very similar so I think there is merit in exploring reuse.

@CBala21 @kv2019i @ujfalusi @bardliao what do you think?

Reply:
I'm not sure whether its ok to mix Cirrus and Realtek in single driver. will it cause any conflict in selection of kconfig?
Also here HDAC HDMI support added but its not present on RT1308. anyhow its not big issue to add top on it.

Reply:
> I'm not sure whether its ok to mix Cirrus and Realtek in single driver. will it cause any conflict in selection of kconfig? 

That should be fine. We already mix Maxim and Realtek codecs in [sof_sdw.c](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw.c)

Reply:
I think @CBala21 's concern was if they only want to add the Realtek RT1308 dependency. That could be done with two entries in the Kconfig selecting common code where we have a IS_ENABLED() test. We've done this in the past with the DA7219 stuff selecting common parts in this folder already.

My point is really that more than half of the code is common.

Reply:
I'll try to add cs35l41 support to the rt1308 machine driver and have ODM to test it.